packages/grid/x-grid-modules/src/hooks/root/useEvents.ts

```diff
@@ -0,0 +1,208 @@
+import * as React from 'react';
+import { useLogger } from '../utils/useLogger';
+import {
+  CELL_CLICK,
+  CLICK_EVENT,
+  COL_RESIZE_START,
+  COL_RESIZE_STOP,
+  COLUMN_HEADER_CLICK,
+  UNMOUNT,
+  KEYDOWN_EVENT,
+  KEYUP_EVENT,
+  RESIZE,
+  ROW_CLICK,
+  HOVER_EVENT,
+  CELL_HOVER,
+  ROW_HOVER,
+  COLUMN_HEADER_HOVER,
+} from '../../constants/eventsConstants';
+import { GridOptions, ApiRef, CellParams, ColParams, RowParams } from '../../models';
+import {
+  CELL_CSS_CLASS,
+  HEADER_CELL_CSS_CLASS,
+  ROW_CSS_CLASS,
+} from '../../constants/cssClassesConstants';
+import {
+  findParentElementFromClassName,
+  getDataFromElem,
+  getFieldFromHeaderElem,
+  getIdFromRowElem,
+  isCell,
+  isHeaderCell,
+} from '../../utils/domUtils';
+import { useApiMethod } from './useApiMethod';
+import { useApiEventHandler } from './useApiEventHandler';
+import { buildCellParams, buildRowParams } from '../../utils/paramsUtils';
+import {EventsApi} from "../../models/api/eventsApi";
+
+export function useEvents(
+  gridRootRef: React.RefObject<HTMLDivElement>,
+  options: GridOptions,
+  apiRef: ApiRef,
+): void {
+  //We use the isResizingRef to prevent the click on column header when the user is resizing the column
+  const isResizingRef = React.useRef(false);
+  const logger = useLogger('useEvents');
+
+  const getHandler = React.useCallback(
+    (name: string) => (...args: any[]) => apiRef.current!.emitEvent(name, ...args),
+    [],
+  );
+
+  const getEventParams = React.useCallback(
+    (event: any) => {
+      if (event.target == null) {
```

Comment:
Well I don't think it can happen. If it does and we throw it can break the app 🤔

Reply:
If you don't think that it can happen, I think that throwing will help us. No matter what happen, we will win.

- If it throw, we would have learn something new about how the logic behaves. Help make better design decisions later on, or even spot bugs. 
- If it doesn't throw. We would have reduced the combinatory of the logic, reducing the number of cases we need to consider, which mean a simpler logic.